File: biz/model/api/branch_rule.go
package api

type BranchRuleDTO struct {
	ID                uint     `json:"id"`
	Prefix            string   `json:"prefix"`
	DisplayName       string   `json:"display_name"`
	SourceBranches    []string `json:"source_branches"`
	TargetBranches    []string `json:"target_branches"`
	RequireTaskID     bool     `json:"require_task_id"`
	TaskIDPattern     string   `json:"task_id_pattern"`
	AutoDeleteOnMerge bool     `json:"auto_delete_on_merge"`
	AllowDirectPush   bool     `json:"allow_direct_push"`
	RequireCodeReview bool     `json:"require_code_review"`
	SortOrder         int      `json:"sort_order"`
}

type BranchRuleSetDTO struct {
	Enabled        bool           `json:"enabled"`
	Rules          []BranchRuleDTO `json:"rules"`
	Protected      []string       `json:"protected_branches"`
	UseCustomRules bool           `json:"use_custom_rules,omitempty"`
	ScopeType      string         `json:"scope_type,omitempty"`
	LinkedRepos    []LinkedRepoDTO `json:"linked_repos,omitempty"`
}

type BranchValidationResult struct {
	Valid    bool                    `json:"valid"`
	Errors   []BranchValidationError `json:"errors,omitempty"`
	RuleName string                  `json:"rule_name,omitempty"`
}

type BranchValidationError struct {
	Field   string `json:"field"`
	Message string `json:"message"`
}

type RemoteRepoBranchRulesDTO struct {
	ProviderConfigID uint            `json:"provider_config_id"`
	PlatformOwner    string          `json:"platform_owner"`
	PlatformRepo     string          `json:"platform_repo"`
	UseCustomRules   bool            `json:"use_custom_rules"`
	Rules            []BranchRuleDTO `json:"rules"`
	Protected        []string        `json:"protected_branches"`
	LinkedRepos      []LinkedRepoDTO `json:"linked_repos"`
}

File: frontend/src/api/modules/branch-rule.ts
import request from '../request'

export interface BranchRuleDTO {
  id: number
  prefix: string
  display_name: string
  source_branches: string[]
  target_branches: string[]
  require_task_id: boolean
  task_id_pattern: string
  auto_delete_on_merge: boolean
  allow_direct_push: boolean
  require_code_review: boolean
  sort_order: number
}

export interface BranchRuleSetDTO {
  enabled: boolean
  rules: BranchRuleDTO[]
  protected_branches: string[]
}

export interface RemoteRepoBranchRulesDTO {
  provider_config_id: number
  platform_owner: string
  platform_repo: string
  use_custom_rules: boolean
  rules: BranchRuleDTO[]
  protected_branches: string[]
  linked_repos: { id: number; key: string; name: string }[]
}

export interface BranchValidationResult {
  valid: boolean
  errors?: { field: string; message: string }[]
  rule_name?: string
}

export function getBranchRules() {
  return request.get<unknown, BranchRuleSetDTO>('/settings/branch-rules')
}

export function updateBranchRules(data: BranchRuleSetDTO) {
  return request.put<unknown, BranchRuleSetDTO>('/settings/branch-rules', data)
}

export function getRemoteRepoBranchRules(providerId: number, owner: string, repo: string) {
  return request.get<unknown, RemoteRepoBranchRulesDTO>(`/branch-rules/remote-config/${providerId}/${owner}/${repo}`)
}

export function updateRemoteRepoBranchRules(providerId: number, owner: string, repo: string, data: Partial<RemoteRepoBranchRulesDTO>) {
  return request.put<unknown, RemoteRepoBranchRulesDTO>(`/branch-rules/remote-config/${providerId}/${owner}/${repo}`, data)
}

export function validateBranchName(params: { repo_key: string; branch_name: string; base_ref?: string; skip_rules?: boolean }) {
  return request.get<unknown, BranchValidationResult>('/settings/branch-rules/validate', { params })
}
